river/neighbors/knn_regressor.py

```diff
@@ -139,40 +125,34 @@ def predict_one(self, x):
         Returns
         -------
             The prediction.
-
         """
+        # Find the nearest neighbors!
+        nearest = self.nn.find_nearest((x, None), n_neighbors=self.n_neighbors)
 
-        if self.data_window.size == 0:
-            # Not enough information available, return default prediction
+        if not nearest:
             return 0.0
 
-        x_arr = dict2numpy(x)
-
-        dists, neighbor_idx = self._get_neighbors(x_arr)
-        target_buffer = self.data_window.targets_buffer
+        # For each in nearest, call it 'item"
+        # item[0] is the original item (x, y)
+        # item[-1] is the distance
+        # item[1:n-1] are extra we don't use here
 
-        # If the closest neighbor has a distance of 0, then return it's output
-        if dists[0][0] == 0:
-            return target_buffer[neighbor_idx[0][0]]
+        # If the closest distance is 0 (it's the same) return it's output (y)
+        # BUT only if the output (y) is not None.
+        if nearest[0][-1] == 0 and nearest[0][0][1] is not None:
+            return nearest[0][0][1]
 
-        if self.data_window.size < self.n_neighbors:  # Select only the valid neighbors
-            neighbor_vals = [
-                target_buffer[index]
-                for cnt, index in enumerate(neighbor_idx[0])
-                if cnt < self.data_window.size
-            ]
-            dists = [
-                dist for cnt, dist in enumerate(dists[0]) if cnt < self.data_window.size
-            ]
-        else:
-            neighbor_vals = [target_buffer[index] for index in neighbor_idx[0]]
-            dists = dists[0]
+        # Only include neighbors in the sum that are non None
+        neighbor_vals = [n[0][1] for n in nearest if n[0][1] is not None]
 
         if self.aggregation_method == self._MEAN:
             return np.mean(neighbor_vals)
```

Comment:
I would prefer avoiding numpy as much as possible. You'll find that it's actually slower than pure Python on small volumes of data. I would instead use `statistics.mean`